### PR TITLE
Updation to chrome >=45 stop function deprecated resolved

### PR DESCRIPTION
**Pull request by hardeepmehta**

### CODE REVIEW

**Comment by katspaugh** [comment]
Why not just check for existence of `getTracks`?

**Comment by hardeepmehta** [comment]
Yes, you are right. It's the better way.

**Comment by thijstriemstra** [comment]
Can the Chrome-related code be moved to wavesurfer.util, I'd rather not have it in the plugin and that way it can be re-used in other plugins.

File: plugin/wavesurfer.microphone.js [hunk 1]
```diff
@@ -122,8 +122,32 @@
             this.disconnect();
 
             // stop stream from device
-            if (this.stream) {
-                this.stream.stop();
+<<<<<<< HEAD
```
Comment (katspaugh) [inline plugin/wavesurfer.microphone.js]: Looks like a failed merge.

File: plugin/wavesurfer.microphone.js [hunk 2]
```diff
@@ -122,8 +122,32 @@
...
+            if (this.stream && this.stream.getTracks()) {
```
Comment (katspaugh) [inline plugin/wavesurfer.microphone.js]: The check of the stream being not null should be outside of the `getTracks` change.

```
if (this.stream) {
    if (this.stream.getTracks) { // note that this should not be a call
        this.stream.getTracks().forEach(function (stream) {
            stream.stop();
        });
    } else {
        this.stream.stop():
    }
}
```

File: plugin/wavesurfer.microphone.js [hunk 3]
```diff
@@ -122,8 +122,32 @@
...
+                
+                this.stream.getTracks().forEach(function( stream ) {
```
Comment (katspaugh) [inline plugin/wavesurfer.microphone.js]: Wrong spaces:
`function( stream ) {` → `function (stream) {`

**Comment by thijstriemstra** [comment]
See #657

**Pull request closed without merging**